mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -449,7 +450,10 @@ case class ScalaPresentationCompiler(
       item.setDocumentation(hoverContent(None, None, documentation))
     }
 
-    item.setDeprecated(completion.symbols.forall(_.isDeprecated))
+    if (completion.symbols.forall(_.isDeprecated)) {
+      item.setTags((Option(item.getTags).fold(Set.empty[CompletionItemTag])(_.asScala.toSet) ++ Set(CompletionItemTag.Deprecated)).toList.asJava)
```

Comment:
Honestly, we would be sacrificing readability for the sake of possible small issue in the future. It looks a bit scary really.

Reply:
This is why I initially had them in the companion object.